pkgs/applications/science/math/cplex/default.nix

```diff
@@ -0,0 +1,75 @@
+{ stdenv, coreutils, requireFile, patchelf, makeWrapper, openjdk, gtk2, xorg, glibcLocales, releasePath ? null }:
+
+# To use this package, you need to download your own cplex installer from IBM
+# and override the releasePath attribute to point to the location of the file.  
+#
+# Note: cplex creates an individual build for each license which screws
+# somewhat with the use of functions like requireFile as the hash will be
+# different for every user.
+
+stdenv.mkDerivation rec {
+  name = "cplex-${version}";
+  version = "128";
+  
+  src =
+    if builtins.isNull releasePath then
+      throw ''
+        This nix expression requires that the cplex installer is already
+        downloaded to your machine. Get it from IBM and override the
+        releasePath attribute to point to the location of the file. 
+      ''
+    else
+      releasePath;
+
+  nativeBuildInputs = [ coreutils makeWrapper ];
+  buildInputs = [ openjdk gtk2 xorg.libXtst glibcLocales ];
+
+  phases = "patchPhase buildPhase installPhase fixupPhase";
+
+  patchPhase = ''
+    sed -e 's|/usr/bin/tr"|tr"         |' $src > $name
```

Comment:
Use `sed -i` to modify the file inplace, no need to `> $name`

Reply:
But $src is the original file, I don't want to modify that one. (The other option would be to first copy the file in the unpack phase...)

Reply:
With the cp in the unpackPhase, sed -i does the job. (BTW, I did not use the builtin substituteInPlace because it is way too slow on such a large file).